iris-mpc-cpu/src/protocol/binary.rs

```diff
@@ -413,6 +413,138 @@ pub(crate) async fn bit_inject_ot_2round(
     Ok(res)
 }
 
+/// This function splits a bit shared additively over F_2 into three shares that need to be combined using the arithmetic xor function
+/// This is a completely local function, `session` is only used to determine the current party id.
+pub(crate) fn split_for_arithmetic_xor(
```

Comment:
You might want to reuse `a2b_pre` here.

Reply:
I saw that but that: 1) would do the party matching for each share instead of once, and 2) this converts bit shares directly to u32 shares instead.